test/integration/scripts/update_config.js

```
@@ -0,0 +1,636 @@
+
+		// Act
+		const updatedConfig = getUpdatedConfig(outputPath, inputPath, '1.6.0', '2.0.0', 'devnet');
+
+		// Assert
+		expect(updatedConfig.components.cache.enabled).to.eql(config.cacheEnabled);
+		expect(updatedConfig.modules.http_api.trustProxy).to.eql(config.trustProxy);
+	});
+
+	it('should remove custom config for db.host and db.port if config equals default config', async () => {
+		// Arrange
+		const config = cloneDeep(baseConfig);
+		config.db.host = 'localhost'; // Default: 'localhost'
+		config.db.port = 5432; // Default: 5432
+
+		fs.writeFileSync(inputPath, JSON.stringify(config), 'utf8');
+
+		// Act
+		const updatedConfig = getUpdatedConfig(outputPath, inputPath, '1.6.0', '2.0.0', 'devnet');
+
+		// Assert
+		expect(updatedConfig.components.storage.host).to.eql(undefined);
+		expect(updatedConfig.components.storage.port).to.eql(undefined);
+	});
+
+	it('should detect custom config for forging', async () => {
+		// Arrange
+		const config = cloneDeep(baseConfig);
+		config.forging = {
+			force: true,
+			delegates: [
+				'47c8b3d6a9e418f0920ef58383260bcd04799db150612d4ff6eb399bcd07f216',
+			],
+			access: {
+				whitelist: [
+					'10.10.10.10'
+				]
+			}
+		}; // Default: { false, [], { { ['127.0.0.1'] }} }
+
+		fs.writeFileSync(inputPath, JSON.stringify(config), 'utf8');
+
+		// Act
+		const updatedConfig = getUpdatedConfig(outputPath, inputPath, '1.6.0', '2.0.0', 'devnet');
+
+		// Assert
+		expect(updatedConfig.modules.http_api.forging.access).to.eql(config.forging.access);
+		expect(updatedConfig.modules.chain.forging).to.eql({
+			force: config.forging.force,
+			delegates: config.forging.delegates
+		});
+	});
+
+	it('should detect custom config for peers', async () => {
+		// Arrange
+		const config = cloneDeep(baseConfig);
+		config.peers = {
+			list: [
+				{
+					ip: 'mynode.ip.net',
+					wsPort: '5000',
+				}
+			],
+			access: {
+				blackList: [
+					'11c8b3d6a9e418fff20ef58383260bcd04799db150612d4ff6eb399bcd07f111'
```

this should be an ip address know?

---

if we migrate any value, then the application might crash if the error is not handled.